Updated findfile() to search localpath first

jasmeet0915 opened the pull request:
<!--
Please remove the appropriate section.
For example, if this is a new feature, remove all sections except for the "New feature" section

If this is your first time opening a PR, be sure to check the contribution guide:
https://gazebosim.org/docs/all/contributing
-->

# 🦟 Bug fix

Fixes #1172

## Summary
As the issue above mentions, [this](http://sdformat.org/tutorials?tut=quickstart) tutorial for Garden, is not able to parse the SDF file if it was named `model.sdf`. It works if the file name is changed. This was happening due to the search order used in the [`sdf::findFile()`](https://github.com/gazebosim/sdformat/blob/3fdc0a1fe7026fbc1f004de018daf83839a2914e/src/SDF.cc#L61) function. The used search order was causing the `findFile()` to find the [`model.sdf`](https://github.com/gazebosim/sdformat/blob/3fdc0a1fe7026fbc1f004de018daf83839a2914e/sdf/1.10/model.sdf) from the install path instead of the one present locally. Since the found `model.sdf` is actually the SDF spec file for the `<model>` and it is not a valid SDFormat file, it caused the script to throw a not valid error. 
This PR changes the search order used in the `findFile()` function by searching for the local path (current directory) first and solves the issue.

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Updated documentation (as needed)
- [x] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [x] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

ahcorde commented on src/SDF.cc:
```diff
@@ -97,6 +97,17 @@ std::string findFile(const std::string &_filename, bool _searchLocalPath,
+  // First check the local path, if the flag is set.
+  if (_searchLocalPath)
+  {
+    std::string path = sdf::filesystem::append(sdf::filesystem::current_path(),
+                                                filename);
```
```suggestion
    std::string path = sdf::filesystem::append(sdf::filesystem::current_path(),
                                               filename);
```

jasmeet0915 replied:
@ahcorde I have made the suggested changes in the latest commit

scpeters reviewed (approved):
this makes sense to me

scpeters commented:
> this makes sense to me

though we mention the change in the Migration guide

azeey commented:
Can you add a test that creates a temporary `model.sdf` in the current directory and verifies that `sdf::readFile` reads the correct file? Also, could you address @scpeters's request to add the change to the Migration guide?

jasmeet0915 commented:
@azeey I have added a test that, as suggested, creates a temporary `model.sdf` in the current directory and verifies that the correct path is returned by the `sdf::findFile()`. Also updated the migration guide as @scpeters suggested.

quarkytale reviewed (approved):
Works as expected!
